siobrultech_protocols/gem/protocol.py

```diff
@@ -225,6 +247,85 @@ def __str__(self) -> str:
         return f"Expected state to be {expected_str}; but got {self._actual.name}!"
 
 
+# Argument type of an ApiCall.
+T = TypeVar("T")
+# Return type of an ApiCall response parser.
+R = TypeVar("R")
+
+
+class ApiType(Enum):
```

Comment:
Should probably decorate with [`unique`](https://docs.python.org/3/library/enum.html#enum.unique) as well